### DISCUSSION TITLE
Slow if directory contains link to unreachable network location

### DISCUSSION THREAD

**Discussion by fkling** [starship/starship#1328] [General]
Hello :wave: 

I'm hesitant to file this as a bug (or even file an issue at all) because this is probably a super rare edge case and should just be resolved on my side one way or the other. But I hope it's still useful that I mention this.

## Context

I'm on Linux and have it configured to automount a network location (via samba). Basically this means that if I enter/open a specific directory in my home directory, a specific directory from my NAS will be mounted automatically.

To make things easier for me, my home directory also contains a link to a nested folder on the NAS, something like

    lrwxrwxrwx  1 user user     29 Jun 10 11:00  documents -> nas/home/user/Documents

## The problem

If I'm in my home directory and should the network location not be reachable, starship is quite slow. I assume that's because it tries to follow the link, which makes the OS try to mount the network location, which times out.

## Thoughts

I'm wondering whether it's reasonable to handle this on the starship side somehow? E.g. by having an option to disable following links. Or some other smart way of handling timeouts.

If not, that's fine :slightly_smiling_face: 
Thank you!

**Comment by chipbuster** [comment]
Hey @fkling,

Not an edge case at all! Off the top of my head, I can't think of what would cause this, but issues like this have been unfortunately more common than we'd like--there was one in the early days that was caused by, I think, some paths being linked into an iCloud Drive or something (unfortunately I can no longer find the exact issue).

The timeout issue is being tracked over at #312. If you'd like to open an issue specifically addressing the path-following issue, we'll try to look into that as well.
